.github/workflows/web_app_deploy.yaml

```diff
@@ -0,0 +1,184 @@
+on:
+  workflow_call:
+    inputs:
+      workspace_name:
+        description: The name of the workspace to create the artifact for.
+        type: string
+        required: true
+      target_name:
+        description: "(Optional) The name of the target to create artifact for (default: main)"
```

Comment:
what do you mean here with 'target'?

Reply:
Workspaces can have multiple entry points, `target_name` is the name to the application to deploy. I borrowed the name `target` from other build tools such as `make`, `xmake`, `target`

`package.json` has two ways to specify an entry point
1. `main` field - legacy and simpler, used by old CJS-only projects. It supports only one `entry point` (the "default one")
2. `exports` field - more advanced, that supports **multiple** entries.

`target_name` - in our context - is the name of the `package.json` entry that holds the path of the entry point file.

### Example 1 (single entry point, basic setup)

```json
{
  "name": "my-app",
  "version": "1.0.0",
  "main": "./dist/index.js"
}
```

`target_name` should be omitted or set to `"main"`

### Example 2 (mutliple entry points)

```json
{
  "name": "my-app",
  "version": "1.0.0",
  "exports": {
    "web": "./dist/web.js",
    "func": "./dist/func.js"
  }
}
```

`target_name` should be either `"web"` or `"func"`

---

This workflow will reduce all workspaces in a node.js project with a single `main` field to enhance compatibility (Azure Functions, for example, relies on `main` field to find the entry point)

I chose not to call this input "entry_point" to avoid confusion. Also because it is not the entry point of the code (`./src/index.ts`) but of the application (`./dist/index.js`)

Reply:
this should be documented someway otherwise it's not clear to the user what 'target_name' means in this context

Reply:
I decided to simplify it using only `main` as target, so I removed this input.